src/NuGet.Core/NuGet.Frameworks/FrameworkNameProvider.cs

```diff
@@ -134,6 +134,20 @@ public bool TryGetShortProfile(string frameworkIdentifier, string profile, out s
             return TryConvertOrNormalize(profile, _profilesToShortName, _profileShortToLong, out profileShortName);
         }
 
+        public bool TryGetShortPlatform(string frameworkIdentifier, Version frameworkVersion, string platformIdentifier, out string platformShortName)
```

Comment:
Possible bugs?

* Isn't it a bug that this method doesn't use platform version?  `.NETCoreApp,Version=5.0,Platform=Windows,PlatformVersion=10.0" has a short name of `net5.0-windows10.0`, right?
  * This is really important for pack, because we need to make sure that when we restore, we know what the TFM's platform version is, so we can do correct package-project compatibility checks.
* If `platformIdentifier` is an empty string, shouldn't the method return `false`?
* Public APIs need to validate input, meaning that all 3 input parameters need to be checked that they are not null, and throw `ArgumentNullException` if they are.

If not, this seems like a weird API to me, to be honest. Maybe it's all just nitpicks and can be ignored in the interest of meeting deadlines, although I'm worried it's a public API and therefore would be a breaking change if we modify it later.

* A full canonical TFM should also have a platform version, so it'm surprised this API doesn't have it as an input, even if it's not used. The only scenario where a missing platform version is allowed is a project `TargetFramework`, but as discussed, the .NET SDK is supposed to tell us what the default TPV is, so any time the TFM is already split into TFI, TFV and TPI, we should also know the TPV. This method not using the TPV is an implementation detail, but it's leaking that knowledge by not allowing it as an input parameter.
* When the TFI is not netcoreapp, or when the major version is less than 5, platform isn't supported. But this API is for getting the platform short name, which I imagine is only called when the caller already knows it's a TFM with a platform (netcoreapp >= 5).
* I can't think of scenarios where anyone would want to call this API. Isn't it only internal for `NuGetFramework.GetShortFolderName()`?
  * If we do want to make getting the "short" TPI, would it be better to force the caller to use `NuGetFramework`? That forces callers to treat the canonical TFM as a single object, rather than the separate components, and reduces the duplicate APIs implementing the same thing.

Reply:
I got rid of the API altogether. That should fix it ;^)